### PR TITLE
MCA/COMMON/UCX: fixed build scripts

### PR DESCRIPTION
**Pull request by hoopoepg**
- updated evaluation of UCX lib - used call from UCX v1.3
- updated makefile compilation flags

Signed-off-by: Sergey Oblomov <sergeyo@mellanox.com>

### CODE REVIEW

**Comment by yosefe** [comment]
@ggouaillardet can you pls check if it solves your issue?

**Comment by hoopoepg** [comment]
@yosefe is it ok to test UCX by any newer API instead of ```ucp_cleanup```?

**Comment by yosefe** [comment]
@hoopoepg a newer API should be optional, not mandatory

**Comment by hoopoepg** [comment]
@yosefe *flush_nb could not be used?
it is required fo fix deadlocks (at least flush_nb calls)

**Comment by yosefe** [comment]
could we disable osc and spml component if the is no flush_nb?
e.g compile the MPI part only
@ggouaillardet what do you think?

**Comment by hoopoepg** [comment]
ucx_common will be complicated: we will have to split common UCX module to 2 parts: legacy & new based API, and handle dependencies based on API available

**Comment by hoopoepg** [comment]
@yosefe what is our decision? add preprocessing for UCX version to fallback to older API?

**Comment by hoopoepg** [comment]
ok, added fallback mode

File: config/ompi_check_ucx.m4 [hunk 1]
```diff
@@ -108,6 +108,18 @@ AC_DEFUN([OMPI_CHECK_UCX],[
                                  [AC_DEFINE([HAVE_UCP_TAG_SEND_NBR],[1],
                                             [have ucp_tag_send_nbr()])], [],
                                  [#include <ucp/api/ucp.h>])
+                  AC_CHECK_DECLS([ucp_ep_flush_nb],
+                                 [AC_DEFINE([HAVE_UCP_EP_FLUSH_NB],[1],
```
[the remark this answers is not in the document]
Comment (hoopoepg) [reply config/ompi_check_ucx.m4]: hmmm, never used _DECL_ macro... ok, fixed

**Pull request merged**